src/packaging.rs

```diff
@@ -337,29 +343,38 @@ where
     I: IntoIterator<Item = P>,
     P: AsRef<Path>,
 {
-    let mut lc_map: HashMap<String, Vec<PathBuf>> = HashMap::new();
+    // Convert paths to a vector for parallel processing
+    let paths_vec: Vec<PathBuf> = paths
+        .into_iter()
+        .map(|p| p.as_ref().to_path_buf())
+        .collect();
 
-    for path_ref in paths {
-        let path = path_ref.as_ref();
-        let case_folded = normalize_path_for_comparison(path, true).unwrap_or_else(|err| {
-            tracing::warn!(
-                "Failed to normalize path for comparison: {}: {}",
-                path.display(),
-                err
-            );
-            path.display().to_string().to_lowercase()
-        });
+    // Parallel normalization
+    let normalized_pairs: Vec<(String, PathBuf)> = paths_vec
```

Comment:
I highly doubt this will have a significant impact on performance. The function is tiny!

Please take a look if it makes sense to parallelize these things!